dvc/repo/experiments/show.py

```diff
@@ -81,7 +81,10 @@ def collect_experiment_commit(
     if exp_rev != "workspace" and not (force or param_deps):
         cached_exp = cache.get(exp_rev)
         if cached_exp:
-            if status == ExpStatus.Running:
+            if status == ExpStatus.Running or (
+                isinstance(cached_exp, SerializableExp)
+                and cached_exp.status == ExpStatus.Running.name
```

Comment:
This state is no longer reachable with the change included in this PR to skip caching any experiment with `Running` state, but this check makes sure we invalidate any cache entries generated prior to this bugfix